src/core/listener.c

```diff
@@ -855,10 +880,46 @@ QuicListenerParamGet(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_DOS_MODE_EVENTS:
+
+        if (Buffer == NULL) {
+            return QUIC_STATUS_INVALID_PARAMETER;
+        }
+
+        if (*BufferLength < sizeof(Listener->DosModeEventsEnabled)) {
+            return QUIC_STATUS_BUFFER_TOO_SMALL;
+        }
```

Comment:
Reorder these checks. We allow a NULL buffer to be passed in to query the output length.